src/affexpr.jl

```diff
@@ -181,13 +181,15 @@ function addconstraint(m::Model, c::LinearConstraint)
     end
     return LinConstrRef(m,length(m.linconstr))
 end
-addconstraint(m::Model, c::Array{LinearConstraint}) =
+addconstraint(m::Model, c::AbstractArray{LinearConstraint}) =
     error("The operators <=, >=, and == can only be used to specify scalar constraints. If you are trying to add a vectorized constraint, use the element-wise dot comparison operators (.<=, .>=, or .==) instead")
 
-function addVectorizedConstraint(m::Model, v::Array{LinearConstraint})
-    ret = Array{LinConstrRef}(size(v))
-    for I in eachindex(v)
-        ret[I] = addconstraint(m, v[I])
+function addVectorizedConstraint(m::Model, v::AbstractArray{LinearConstraint})
```

Comment:
Without changing this method, the following fails:
```julia
m = Model()
v = @variable(m, [1:3])
x = OffsetArray(v, -3)
@constraint(m, x .== 0)
```
because there is no `Convert` method that constructs an `Array` from an `OffsetArray` (a conscious design decision in OffsetArrays).